jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/udp/tim/TimReceiver.java

```diff
@@ -52,33 +41,19 @@ public void run() {
             logger.debug("Waiting for UDP TIM packets...");
             socket.receive(packet);
             if (packet.getLength() > 0) {
-               senderIp = packet.getAddress().getHostAddress();
-               senderPort = packet.getPort();
-               logger.debug("Packet received from {}:{}", senderIp, senderPort);
-
-               // Create OdeMsgPayload and OdeLogMetadata objects and populate them
-               OdeAsn1Payload timPayload = super.getPayloadHexString(packet, UperUtil.SupportedMessageTypes.TIM);
-               if (timPayload == null)
-                  continue;
-               OdeTimMetadata timMetadata = new OdeTimMetadata(timPayload);
-
-               // Add header data for the decoding process
-               ZonedDateTime utc = ZonedDateTime.now(ZoneOffset.UTC);
-               String timestamp = utc.format(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"));
-               timMetadata.setOdeReceivedAt(timestamp);
-
-               timMetadata.setOriginIp(senderIp);
-               timMetadata.setRecordType(RecordType.timMsg);
-               timMetadata.setRecordGeneratedBy(GeneratedBy.RSU);
-               timMetadata.setSecurityResultCode(SecurityResultCode.success);
-
-               // Submit JSON to the OdeRawEncodedMessageJson Kafka Topic
-               timPublisher.publish(JsonUtils.toJson(new OdeAsn1Data(timMetadata, timPayload), false),
-                  timPublisher.getOdeProperties().getKafkaTopicOdeRawEncodedTIMJson());
+               
+               String timJson = UdpHexDecoder.buildJsonTimFromPacket(packet);
+               if(timJson != null){
```

Comment:
same comment above

Reply:
Removed comments, and verified null check.